test/unit_test/data_source/test_connector_runner_credential_redaction.py

```
@@ -0,0 +1,291 @@
+
+    def build_dummy_checkpoint(self):
+        return ConnectorCheckpoint(has_more=True)
+
+    def load_from_state(self):
+        # All of these locals intentionally never get read in Python --
+        # the test asserts they appear in the captured log output (for
+        # non-sensitive vars) or are redacted to "<redacted>" (for
+        # sensitive vars). The noqa: F841 marks them as
+        # intentionally-unread; without it ruff misreads the test
+        # intent.
+        access_token = "ya29.a0fake_google_oauth_access_token_value_xyz"  # noqa: F841
```

Ruff (0.16.0)

[error] Possible hardcoded password assigned to: "access_token"

(S105)